src/program.ts

```diff
@@ -4299,22 +4330,23 @@ export class Class extends TypedElement {
 
   /** Creates a buffer suitable to hold a runtime instance of this class. */
   createBuffer(overhead: i32 = 0): Uint8Array {
-    var size = this.nextMemoryOffset + overhead;
-    var buffer = new Uint8Array(this.program.runtimeHeaderSize + size);
-    assert(!this.program.options.isWasm64); // TODO: WASM64, mmInfo is usize
-    // see: std/assembly/rt/common.ts
-    assert(size < (1 << 28));      // 1 bit BUFFERED + 3 bits color
-    var OBJECT = this.program.OBJECTInstance;
-    OBJECT.writeField("mmInfo", size, buffer, 0);
+    var program = this.program;
+    var payloadSize = this.nextMemoryOffset + overhead;
+    var blockSize = program.computeBlockSize(payloadSize, true); // excl. overhead
+    assert(blockSize < (1 << 28)); // PureRC: 1 bit BUFFERED + 3 bits color
```

Comment:
I guess it should be `(1 << 30) - BLOCK_OVERHEAD`:
https://github.com/AssemblyScript/assemblyscript/blob/master/std/assembly/rt/common.ts#L37